components/feature/prompts/src/test/java/mozilla/components/feature/prompts/PromptFeatureTest.kt

```diff
@@ -836,6 +834,70 @@ class PromptFeatureTest {
         assertTrue(onDismissCalled)
     }
 
+    @Test
+    fun `dialog will never be dismissed if page load does not occur`() {
+        val feature = spy(PromptFeature(activity = mock(), store = store, fragmentManager = fragmentManager, shareDelegate = mock()) { })
+        feature.start()
+
+        val shareRequest = PromptRequest.Share(
+            ShareData("Title", "Text", null),
+            onSuccess = {},
+            onFailure = {},
+            onDismiss = {}
+        )
+        store.dispatch(ContentAction.UpdatePromptRequestAction(tabId, shareRequest)).joinBlocking()
+
+        verify(feature, times(0)).activePrompt
+    }
+
+    @Test
+    fun `dialog will be dismissed if progress reaches 90%`() {
+        val feature = spy(PromptFeature(activity = mock(), store = store, fragmentManager = fragmentManager, shareDelegate = mock()) { })
+        feature.start()
+
+        val shareRequest = PromptRequest.Share(
+            ShareData("Title", "Text", null),
+            onSuccess = {},
+            onFailure = {},
+            onDismiss = {}
+        )
+        store.dispatch(ContentAction.UpdatePromptRequestAction(tabId, shareRequest)).joinBlocking()
+
+        store.dispatch(ContentAction.UpdateProgressAction(tabId, 0)).joinBlocking()
+        store.dispatch(ContentAction.UpdateProgressAction(tabId, 10)).joinBlocking()
+        store.dispatch(ContentAction.UpdateProgressAction(tabId, 11)).joinBlocking()
+        store.dispatch(ContentAction.UpdateProgressAction(tabId, 28)).joinBlocking()
+        store.dispatch(ContentAction.UpdateProgressAction(tabId, 32)).joinBlocking()
+        store.dispatch(ContentAction.UpdateProgressAction(tabId, 49)).joinBlocking()
+        store.dispatch(ContentAction.UpdateProgressAction(tabId, 60)).joinBlocking()
+        store.dispatch(ContentAction.UpdateProgressAction(tabId, 89)).joinBlocking()
+        store.dispatch(ContentAction.UpdateProgressAction(tabId, 90)).joinBlocking()
+
+        // dismiss + clear
+        verify(feature, times(2)).activePrompt
```

Comment:
I couldn't figure out any better ways to sense `dismiss` calls in these tests.  I'm open to other options, if anyone has any ideas.